components/script/script_thread.rs

```diff
@@ -3176,11 +3178,13 @@ impl ScriptThread {
             self.timer_event_chan.clone(),
         );
 
-        let origin = if final_url.as_str() == "about:blank" {
-            incomplete.origin.clone()
-        } else {
-            MutableOrigin::new(final_url.origin())
-        };
+        // srcdoc document's origin must share creator document's origin - added OR condition
+        let origin =
+            if final_url.as_str() == "about:blank" || incomplete.url.as_str() == "about:srcdoc" {
```

Comment:
There should be no difference in behaviour using `final_url` here, so let's do that instead.